src/util/prolog/prolog.test.ts

```diff
@@ -177,4 +177,23 @@ describe("prologStringManipulation Module", () => {
 		expect(results).toEqual(assets.expectedNF);
 	});
 
+	test("fixVariableShadowingInImport() renames predicates to have no predicate name shadowing", async () => {
+		const results = PSM.fixVariableShadowingInImport("male(bob).", "male(frank).");
+		expect(results).toEqual("male_1(frank).");
+	});
+
+	test("fixVariableShadowingInImport() increases the identifier on overlap", async () => {
+		const results = PSM.fixVariableShadowingInImport("male(bob). male_1(karl).", "male(frank).");
+		expect(results).toEqual("male_2(frank).");
+	});
+
+	test("fixVariableShadowingInImport() renames multiple predicates", async () => {
+		const results = PSM.fixVariableShadowingInImport("male(bob). male_1(karl).", "male(frank). male(hannes).");
+		expect(results).toEqual("male_2(frank). male_2(hannes).");
+	});
+
+	test("fixVariableShadowingInImport() work even with parameters containing braces", async () => {
+		const results = PSM.fixVariableShadowingInImport("male(b(ob)).", "male(ha(n(ne)s)).");
+		expect(results).toEqual("male_1(ha(n(ne)s)).");
+	});
```

Comment:
Could you give each function it's individual "describe" (as in using describe-cascades)? That makes it more readable, since you see, that some amount of tests all together test one function. Also within the tests for one function you could go for "describes" that group the tests by error-/edge-/normal cases.

Reply:
To group by functions totally make sense which I will implement. For different cases I will think about how to structur that. I should also split the tests over more files since they refere to different functionalities by now.

Reply:
I will revise the test cases later actually. I wanna work with these changes to make diproche run soon.